node/core/approval-voting/src/lib.rs

```diff
@@ -1936,6 +1936,29 @@ async fn launch_approval(
 	let (a_tx, a_rx) = oneshot::channel();
 	let (code_tx, code_rx) = oneshot::channel();
 
+	// The background future returned by this function may
+	// be dropped before completing. This guard is used to ensure that the approval
+	// work is correctly counted as stale even if so.
+	struct StaleGuard(Option<Metrics>);
```

Comment:
Nit: Not sure why we need option here. `forget_with` seems to be mutually exclusive with `drop`.